src/lib/int.ts

```diff
@@ -0,0 +1,391 @@
+      .rangeCheckHelper(UInt64.NUM_BITS)
+      .equals(xMinusY.neg());
+    xMinusYFits.or(yMinusXFits).assertEquals(true);
+    // x <= y if y - x fits in 64 bits
+    return yMinusXFits;
+  }
+
+  assertLte(y: UInt64) {
+    let yMinusX = argToField('UInt64.lt', y).sub(this.value).seal();
+    yMinusX.rangeCheckHelper(UInt64.NUM_BITS).assertEquals(yMinusX);
+  }
+
+  lt(y: UInt64): Bool {
+    return this.lte(y).and(this.value.equals(y.value).not());
+  }
+
+  assertLt(y: UInt64) {
+    this.lt(y).assertEquals(true);
+  }
+
+  gt(y: UInt64): Bool {
+    return y.lt(this);
+  }
+
+  assertGt(y: UInt64) {
+    y.assertLt(this);
+  }
+}
+
+export class UInt32 extends CircuitValue {
+  @prop value: Field;
+
+  static get zero(): UInt32 {
+    return new UInt32(Field.zero);
+  }
+
+  constructor(value: Field) {
+    super();
+    this.value = value;
+  }
+
+  toString(): string {
+    return this.value.toString();
+  }
+
+  static check(x: UInt32) {
+    let actual = x.value.rangeCheckHelper(32);
+    actual.assertEquals(x.value);
+  }
+
+  static fromNumber(x: number): UInt32 {
+    return new UInt32(argToField('UInt32.fromNumber', x));
+  }
+
+  static NUM_BITS = 32;
+
+  static MAXINT(): UInt32 {
+    return new UInt32(Field.fromJSON(((1n << 32n) - 1n).toString()) as Field);
+  }
+
+  divMod(y: UInt32 | number): [UInt32, UInt32] {
+    let x = this.value;
+    let y_ = argToField('UInt32.div', y);
+
+    if (this.value.isConstant() && y_.isConstant()) {
+      let xn = BigInt(x.toString());
+      let yn = BigInt(y_.toString());
+      let q = xn / yn;
+      let r = xn - q * yn;
+      return [
+        new UInt32(new Field(q.toString())),
+        new UInt32(new Field(r.toString())),
+      ];
+    }
+
+    y_ = y_.seal();
+
+    let q = Circuit.witness(
+      Field,
+      () => new Field((BigInt(x.toString()) / BigInt(y_.toString())).toString())
+    );
+
+    q.rangeCheckHelper(UInt32.NUM_BITS).assertEquals(q);
+
+    // TODO: Could be a bit more efficient
+    let r = x.sub(q.mul(y_)).seal();
+    r.rangeCheckHelper(UInt32.NUM_BITS).assertEquals(r);
+
+    let r_ = new UInt32(r);
+    let q_ = new UInt32(q);
+
+    r_.assertLt(new UInt32(y_));
+
+    return [q_, r_];
+  }
+
+  div(y: UInt32 | number): UInt32 {
+    const dm = this.divMod(y);
+    return dm[0];
+  }
+
+  mod(y: UInt32 | number): UInt32 {
+    return this.divMod(y)[1];
+  }
+
+  mul(y: UInt32 | number): UInt32 {
+    let z = this.value.mul(argToField('UInt32.mul', y));
+    z.rangeCheckHelper(UInt32.NUM_BITS).assertEquals(z);
+    return new UInt32(z);
+  }
+
+  add(y: UInt32 | number): UInt32 {
+    let z = this.value.add(argToField('UInt32.add', y));
+    z.rangeCheckHelper(UInt32.NUM_BITS).assertEquals(z);
+    return new UInt32(z);
+  }
+
+  sub(y: UInt32 | number): UInt32 {
+    let z = this.value.sub(argToField('UInt32.sub', y));
+    z.rangeCheckHelper(UInt32.NUM_BITS).assertEquals(z);
+    return new UInt32(z);
+  }
+
+  lte(y: UInt32): Bool {
```

Comment:
iirc we had decided to use the long-form version of these names (e.g. `lessThanOrEqual`), similar to Jest, so that IDE fuzzy matching will work for either situation (i.e. a developer could type `lessTh...` or `lte` and still get the IDE suggestion)

Reply:
Yes! Will try to find time this or next week, by now I can handle changing the Ocaml-side API